src/cargo/sources/git/oxide.rs

```diff
@@ -120,7 +107,88 @@ pub fn with_retry_and_progress(
             }
         });
         network::with_retry(config, || {
-            cb(&repo, &program_aborts_on_interrupt, &mut progress)
+            let res = cb(&repo_path, &program_aborts_on_interrupt, &mut progress);
+            amend_authentication_hints(res)
         })
     })
 }
+
+fn amend_authentication_hints(res: CargoResult<()>) -> CargoResult<()> {
+    match res {
+        Ok(()) => Ok(()),
+        Err(mut err) => {
```

Comment:
nit: Since this review has taken long enough:
```
let Err(mut err) = res else {
    return Ok()));
};
```

For me personally, having a `match` at the top
- Causes needless indentation which makes tracking scopes harder
- I find I prefer my indentation to reflect my business logic and 
not any bookkeeping